gateways/rocketgate/rocketgate.go

```diff
@@ -4,42 +4,126 @@ import (
 	"net/http"
 
 	"github.com/BoltApp/sleet"
+	"github.com/BoltApp/sleet/common"
+	"github.com/BoltApp/sleet/rocketgate-sdk/response"
+	"github.com/BoltApp/sleet/rocketgate-sdk/service"
 )
 
 // RocketgateClient represents an HTTP client and the associated authentication information required for
 // making an API request.
 type RocketgateClient struct {
-	httpClient      *http.Client
+	testMode 			bool
+	merchantID  		string
+	merchantPassword	string
+	merchantAccount 	*string
+	httpClient  		*http.Client
 }
 
 // NewClient creates a Rocketgate client
-func NewClient(httpClient *http.Client) *RocketgateClient {
-	return NewWithHttpClient(httpClient)
+func NewClient(env common.Environment, merchantID string, merchantPassword string, merchantAccount *string, httpClient *http.Client) *RocketgateClient {
+	return NewWithHttpClient(env, merchantID, merchantPassword, merchantAccount, httpClient)
 }
 
 // NewWithHttpClient creates an Rocketgate client for custom behavior
-func NewWithHttpClient(httpClient *http.Client) *RocketgateClient {
+func NewWithHttpClient(env common.Environment, merchantID string, merchantPassword string, merchantAccount *string, httpClient *http.Client) *RocketgateClient {
 	return &RocketgateClient{
-		httpClient:  httpClient,
+		testMode:			rocketgateTestMode(env),
+		merchantID:			merchantID,
+		merchantPassword:	merchantPassword,
+		merchantAccount: 	merchantAccount,
+		httpClient:			httpClient,
 	}
 }
 
 // Authorize a transaction. This transaction must be captured to receive funds
 func (client *RocketgateClient) Authorize(request *sleet.AuthorizationRequest) (*sleet.AuthorizationResponse, error) {
-	return nil, nil
+	gatewayService := service.NewGatewayService()
+	gatewayResponse := response.NewGatewayResponse()
+	gatewayRequest := buildAuthRequest(client.merchantID, client.merchantPassword, client.merchantAccount, request)
+
+	gatewayService.SetTestMode(client.testMode)
+
+	if !gatewayService.PerformAuthOnly(gatewayRequest, gatewayResponse) {
```

Comment:
Yeah, from https://help.rocketgate.com/support/solutions/articles/28000018238-gatewayservice, it mentioned it will return false for errors and declines I think 

`
A boolean value indicating whether the transaction was performed successfully and accepted by the banking institution.  If a “true” value is returned, the transaction completed successfully and the response contains details include the transaction ID, auth-code, etc.  If a “false” value is returned, the transaction failed and the response contains an explanation of the failure.
`

Then we can know the reason based on reason_code or response_code: https://help.rocketgate.com/support/solutions/articles/28000018169-gatewayresponse-error-decline-codes

Reply:
Wow what an interesting error handling mechanism from them. Nice job figuring out how to work with it.